services/factorio-mod-portal/factorio-mod-portal.service.js

```diff
@@ -0,0 +1,198 @@
+import Joi from 'joi'
+import { BaseJsonService } from '../index.js'
+import { age } from '../color-formatters.js'
+import { formatDate } from '../text-formatters.js'
+import { renderDownloadsBadge } from '../downloads.js'
+import { renderVersionBadge } from '../version.js'
+
+const schema = Joi.object({
+  downloads_count: Joi.number().required(),
+  releases: Joi.array()
+    .items(
+      Joi.object({
+        version: Joi.string().required(),
+        released_at: Joi.string().required(),
+        info_json: Joi.object({
+          factorio_version: Joi.string().required(),
+        }).required(),
+      })
+    )
+    .min(1)
+    .required(),
+}).required()
+
+const queryParamSchema = Joi.object({
+  range: Joi.equal(''),
+}).required()
+
+// Factorio Mod portal API
+// @see https://wiki.factorio.com/Mod_portal_API
+class BaseFactorioModPortalService extends BaseJsonService {
+  async fetch({ modName }) {
+    const { releases, downloads_count } = await this._requestJson({
+      schema,
+      url: `https://mods.factorio.com/api/mods/${modName}`,
+      errorMessages: {
+        404: 'mod not found',
+      },
+    })
+
+    return {
+      downloads_count,
+      first_release: releases[0],
+      latest_release: releases[releases.length - 1],
```

Comment:
Do we have complete confidence this array is sorted by version? The docs aren't completely clear.

Reply:
I couldn't find in the docs or anywhere else that explicitly says it's
 sorted by version, but I made a small script to run through every mod listed and can confirm that the `releases` array is always sorted by version.